pincer/client.py

```diff
@@ -586,22 +586,62 @@ async def payload_event_handler(self, _, payload: GatewayDispatch):
         """
         await self.process_event("payload", payload)
     
-    async def create_guild(self, **kwargs) -> Guild:
-        """|coro|
-
-        Creates a guild.
+    @overload
+    async def create_guild(
+        self,
+        *,
+        name: str,
+        region: APINullable[Optional[str]] = None,
+        icon: Optional[str] = None,
+        verification_level: Optional[int] = None,
+        default_message_notifications: Optional[int] = None,
+        explicit_content_filter: Optional[int] = None,
+        roles: Optional[List[Role]] = None,
+        channels: Optional[List[Channel]] = None,
+        afk_channel_id: Optional[Snowflake] = None,
+        afk_timeout: Optional[int] = None,
+        system_channel_id: Optional[Snowflake] = None,
+        system_channel_flags: Optional[int] = None
+    ) -> Guild:
+        """Creates a guild.
 
         Parameters
         ----------
-        \\*\\*kwargs
-            The parameters for the guild.
-        
+        name : :class:`str`
+            Name of the guild (2-100 characters)
+        region : APINullable[Optional[:class:`str`]]
+            Voice region id (deprecated) |default| :data:`None`
+        icon : Optional[:class:`str`]
+            base64 128x128 image for the guild icon |default| :data:`None`
+        verification_level : Optional[:class:`int`]
+            Verification level |default| :data:`None`
+        default_message_notifications : Optional[:class:`int`]
+            Default message notification level |default| :data:`None`
+        explicit_content_filter : Optional[:class:`int`]
+            Explicit content filter level |default| :data:`None`
+        roles : Optional[List[:class:`~pincer.objects.guild.role.Role`]]
+            New guild roles |default| :data:`None`
+        channels : Optional[List[:class:`~pincer.objects.guild.channel.Channel`]]
+            New guild's channels |default| :data:`None`
+        afk_channel_id : Optional[:class:`~pincer.utils.snowflake.Snowflake`]
+            ID for AFK channel |default| :data:`None`
+        afk_timeout : Optional[:class:`int`]
+            AFK timeout in seconds |default| :data:`None`
+        system_channel_id : Optional[:class:`~pincer.utils.snowflake.Snowflake`]
+            The ID of the channel where guild notices such as welcome
+            messages and boost events are posted |default| :data:`None`
+        system_channel_flags : Optional[:class:`int`]
+            System channel flags |default| :data:`None`
+
         Returns
         -------
         :class:`~pincer.objects.guild.guild.Guild`
-            The guild that was created.
+            The created guild
         """
-        g = await self.http.post("guilds", data=kwargs)
+        ...
+    
+    async def create_guild(self, name: str, **kwargs) -> Guild:
+        g = await self.http.post("guilds", data={"name": name, **kwargs})
         return await self.get_guild(g['id'])
```

Comment:
why do `Client.get_guild` here? why not use the guild object that comes from the `create_guild`?

Reply:
Ask @Sigmanificient

Reply:
Cause then it have `_http = None`

Reply:
That’s what `create_client_dict` is for.